examples/defaulter-gen/generators/defaulter.go

```diff
@@ -473,12 +577,22 @@ func (c *callTreeForType) build(t *types.Type, root bool) *callNode {
 				child.elem = true
 			}
 			parent.children = append(parent.children, *child)
+		} else if member := populateDefaultValue(nil, t.Elem, "json:\"name,omitempty\"", t.Elem.CommentLines); member != nil {
+			member.index = true
+			member.markerOnly = true
+			parent.children = append(parent.children, *member)
 		}
+
 	case types.Map:
 		if child := c.build(t.Elem, false); child != nil {
 			child.key = true
 			parent.children = append(parent.children, *child)
+		} else if member := populateDefaultValue(nil, t.Elem, "json:\"name,omitempty\"", t.Elem.CommentLines); member != nil {
```

Comment:
Rather than passing a made-up tag here, I would pass-in the information that you care about in the tags. If I read `populateDefaultValue` properly, you only use the tag to know if the field is omitempty. Can you pass omitempty as a boolean instead of the tag?

Reply:
This was meant as a workaround for https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1929-built-in-default#string-maps, but since that case should be invalid we can remove this code path.